File: 2024-2/0682/unidad06.d_Web_04/SoftPub/SoftPubPersonaBussiness/SeccionBo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using SoftPubPersonaControler.DAO;
using SoftPubPersonaControler.DAOImpl;
using SoftPubPersonaModel;

namespace SoftPubPersonaBussiness
{
    public class SeccionBo
    {
        private SeccionDAO seccionDAO;

        public SeccionBo()
        {
            this.seccionDAO = new SeccionDAOImpl();
        }

        public int insertar(string idSeccion, string nombre_seccion, string departamento)
        {
            Seccion seccion = new Seccion(idSeccion, nombre_seccion, departamento);
            return this.seccionDAO.insertar(seccion);
        }

        public int modificar(string idSeccion, string nombre_seccion, string departamento)
        {
            Seccion seccion = new Seccion(idSeccion, nombre_seccion, departamento);
            return this.seccionDAO.modificar(seccion);
        }

        public int eliminar(string idSeccion)
        {
            Seccion seccion = new Seccion();
            seccion.IdSeccion = idSeccion;
            return this.seccionDAO.eliminar(seccion);
        }

        public BindingList<Object> listarTodos()
        {
            return this.seccionDAO.listarTodos();
        }

        public Seccion obtenerPorId(string idSeccion)
        {
            return this.seccionDAO.obtenerPorId(idSeccion);
        }        
    }
}

File: 2024-2/0682/unidad07.b - Programación distribuida con BD/SoftPub_Back-end/SoftPubPersonaBO/src/pe/edu/pucp/softpub/persona/bo/SeccionBO.java
package pe.edu.pucp.softpub.persona.bo;

import java.util.ArrayList;
import pe.edu.pucp.softpub.persona.dao.SeccionDAO;
import pe.edu.pucp.softpub.persona.daoImpl.SeccionDAOImpl;
import pe.edu.pucp.softpub.persona.model.Seccion;

public class SeccionBO {

    private final SeccionDAO seccionDAO;

    public SeccionBO() {
        this.seccionDAO = new SeccionDAOImpl();
    }
    
    public Integer insertar(String idSeccion, String nombre_seccion, String departamento){
        Seccion seccion = new Seccion(idSeccion,  nombre_seccion,  departamento);
        return seccionDAO.insertar(seccion);
    }
    
    public Integer modificar(String idSeccion, String nombre_seccion, String departamento){
        Seccion seccion = new Seccion(idSeccion,  nombre_seccion,  departamento);
        return seccionDAO.modificar(seccion);
    }
    
    public Integer eliminar(String idSeccion){
        Seccion seccion = new Seccion();
        seccion.setIdSeccion(idSeccion);
        return seccionDAO.eliminar(seccion);
    }
    
    public ArrayList<Seccion> listarTodos(){
        return seccionDAO.listarTodos();
    }
    
    public Seccion obtenerPorId(String idSeccion){
        return seccionDAO.obtenerPorId(idSeccion);
    }       
}
